File: lib_network/src/main/java/com/android/sdk/net/core/exception/ServerErrorException.java
package com.android.sdk.net.core.exception;

import androidx.annotation.NonNull;

/**
 * @author Ztiany
 */
public class ServerErrorException extends RuntimeException {

    private final int mErrorType;

    public static final int DATA_PARSE_ERROR = 1;
    public static final int EMPTY_SERVER_DATA = 2;

    /**
     * @param errorType {@link #DATA_PARSE_ERROR} or {@link #EMPTY_SERVER_DATA}
     */
    public ServerErrorException(int errorType, @NonNull Exception cause) {
        super(cause);
        mErrorType = errorType;
    }

    /**
     * @param errorType {@link #DATA_PARSE_ERROR} or {@link #EMPTY_SERVER_DATA}
     */
    public ServerErrorException(int errorType) {
        mErrorType = errorType;
    }

    public int getErrorType() {
        return mErrorType;
    }

    @NonNull
    @Override
    public String toString() {
        String string = super.toString();
        return string + "ErrorType = " + transform();
    }

    private String transform() {
        return switch (mErrorType) {
            case DATA_PARSE_ERROR -> "DATA_PARSE_ERROR";
            case EMPTY_SERVER_DATA -> "EMPTY_SERVER_DATA";
            default -> "Unknown";
        };
    }

}

File: lib_network/src/main/java/com/android/sdk/net/core/exception/ServerErrorException.kt
package com.android.sdk.net.core.exception

/**
 * @author Ztiany
 */
class ServerErrorException : RuntimeException {

    val errorType: Int

    /**
     * @param errorType [.DATA_PARSE_ERROR] or [.EMPTY_SERVER_DATA]
     */
    constructor(errorType: Int, cause: Exception) : super(cause) {
        this.errorType = errorType
    }

    /**
     * @param errorType [.DATA_PARSE_ERROR] or [.EMPTY_SERVER_DATA]
     */
    constructor(errorType: Int) {
        this.errorType = errorType
    }

    override fun toString(): String {
        val string = super.toString()
        return string + "ErrorType = " + name()
    }

    private fun name(): String {
        return when (errorType) {
            DATA_PARSE_ERROR -> "DATA_PARSE_ERROR"
            EMPTY_SERVER_DATA -> "EMPTY_SERVER_DATA"
            else -> "Unknown"
        }
    }

    companion object {
        const val DATA_PARSE_ERROR: Int = 1
        const val EMPTY_SERVER_DATA: Int = 2
    }

}